src/main/java/io/kestra/plugin/googleworkspace/calendar/EventCreatedTrigger.java

```diff
@@ -0,0 +1,492 @@
+    )
+    protected Property<String> organizerEmail;
+
+    @Schema(
+        title = "Event status filter",
+        description = "Only events with this status will trigger executions."
+    )
+    protected Property<EventStatus> eventStatus;
+
+    @Schema(
+        title = "The polling interval",
+        description = "How frequently to check for new events. Must be at least PT1M (1 minute)."
+    )
+    @Builder.Default
+    protected Duration interval = Duration.ofMinutes(5);
+
+    @Schema(
+        title = "Maximum number of events to process per poll",
+        description = "Limits the number of new events processed in a single poll to avoid overwhelming the system. " +
+            "Valid range: 1-" + MAX_EVENTS_PER_POLL + " (Google Calendar API maximum). Default is 100."
+    )
+    @Builder.Default
+    protected Property<Integer> maxEventsPerPoll = Property.ofValue(100);
+
+    @Override
+    public Duration getInterval() {
+        return this.interval;
+    }
+
+    @Override
+    public Property<Integer> getReadTimeout() {
+        return Property.ofValue(120);
+    }
+
+    @Override
+    public Property<List<String>> getScopes() {
+        return Property.ofValue(List.of("https://www.googleapis.com/auth/calendar"));
+    }
+
+    @Override
+    public Optional<Execution> evaluate(ConditionContext conditionContext, TriggerContext context) throws Exception {
+        RunContext runContext = conditionContext.getRunContext();
+        Logger logger = runContext.logger();
+
+        validateConfiguration(runContext);
+
+        // Create Calendar connection
+        Calendar calendarService = this.connection(runContext);
+
+        // Find the time window for checking new events
+        Instant lastCreatedTime = context.getNextExecutionDate() != null
+            ? context.getNextExecutionDate().toInstant().minus((TemporalAmount) this.interval)
+            : Instant.now().minus((TemporalAmount) this.interval);
+
+        logger.debug("Checking for events created after: {}", lastCreatedTime);
+
+        List<String> calendarsToMonitor = getCalendarsToMonitor(runContext);
+        
+        List<EventMetadata> allNewEvents = new ArrayList<>();
+
+        //Check every calendar for new events
+        for (String calId : calendarsToMonitor) {
+            try {
+                List<EventMetadata> newEvents = checkCalendarForNewEvents(
+                    calendarService, runContext, calId, lastCreatedTime, logger
+                );
+                allNewEvents.addAll(newEvents);
+            } catch (Exception e) {
+                logger.warn("Error checking calendar '{}': {}", calId, e.getMessage());
+            }
+        }
+
+        if (allNewEvents.isEmpty()) {
+            return Optional.empty();
+        }
+
+        logger.info("Found {} new event(s)", allNewEvents.size());
+
+        // Create an execution with the first event as variables
+        EventMetadata firstEvent = allNewEvents.get(0);
```

Comment:
Why having a list of events in the Output to just return a single first event here? You should return a list of events.